include/modem_info.h

```diff
@@ -117,7 +182,19 @@ enum at_param_type modem_info_type_get(enum modem_info info);
  *         successful.
  *         Otherwise, a (negative) error code is returned.
  */
-int modem_info_json_string_get(char *buf);
+int modem_info_json_string_encode(struct modem_param_info *modem_param,
```

Comment:
I think you should remove the `json` keyword here and declae this function as `extern`. Then you could select at runtime which encoding you want to use.

Reply:
I don't agree. 

Right now the library only supports JSON encoding, and it is not in the pipeline to add any different encoding types to my knowledge. I welcome different encoding types of course, but I don't think it fits the scope of this PR to change the API based on this notion, as it simply adds complexity.